src/domain/sample_slots.rs

```diff
@@ -0,0 +1,134 @@
+//! Handles backups of the volca's sample memory slot layout
+use serde::de::{Deserialize, Deserializer, MapAccess, Visitor};
+use serde::ser::{Serialize, SerializeMap, Serializer};
+use std::fmt;
+use std::ops::{Index, IndexMut};
+
+extern crate serde_yaml;
```

Comment:
```suggestion
use std::fmt;
use std::ops::{Index, IndexMut};

use serde::de::{Deserialize, Deserializer, MapAccess, Visitor};
use serde::ser::{Serialize, SerializeMap, Serializer};
```

[`extern` not needed](https://doc.rust-lang.org/edition-guide/rust-2018/path-changes.html#no-more-extern-crate).